tests/store/tracking/test_file_store.py

```diff
@@ -2562,3 +2563,100 @@ def test_log_inputs_handles_case_when_no_datasets_are_specified(store):
+
+def test_search_datasets(store):
+    exp_id1 = store.create_experiment("test_search_datasets_1")
+    # Create an additional experiment to ensure we filter on specified experiment
+    # and search works on multiple experiments.
+    exp_id2 = store.create_experiment("test_search_datasets_2")
+
+    run1 = store.create_run(
+        experiment_id=exp_id1,
+        user_id="user",
+        start_time=1,
+        tags=[],
+        run_name=None,
+    )
+    run2 = store.create_run(
+        experiment_id=exp_id1,
+        user_id="user",
+        start_time=2,
+        tags=[],
+        run_name=None,
+    )
+    run3 = store.create_run(
+        experiment_id=exp_id2,
+        user_id="user",
+        start_time=3,
+        tags=[],
+        run_name=None,
+    )
+
+    dataset1 = Dataset(
+        name="name1",
+        digest="digest1",
+        source_type="st1",
+        source="source1",
+        schema="schema1",
+        profile="profile1",
+    )
+    dataset2 = Dataset(
+        name="name2",
+        digest="digest2",
+        source_type="st2",
+        source="source2",
+        schema="schema2",
+        profile="profile2",
+    )
+    dataset3 = Dataset(
+        name="name3",
+        digest="digest3",
+        source_type="st3",
+        source="source3",
+        schema="schema3",
+        profile="profile3",
+    )
+
+    test_tag = [InputTag(key=MLFLOW_DATASET_CONTEXT, value="test")]
+    train_tag = [InputTag(key=MLFLOW_DATASET_CONTEXT, value="train")]
+    eval_tag = [InputTag(key=MLFLOW_DATASET_CONTEXT, value="eval")]
+
+    inputs_run1 = [
+        DatasetInput(dataset1, train_tag),
+        DatasetInput(dataset2, eval_tag),
+    ]
+    inputs_run2 = [
+        DatasetInput(dataset1, train_tag),
+        DatasetInput(dataset2, test_tag),
+    ]
+    inputs_run3 = [DatasetInput(dataset3, train_tag)]
+
+    store.log_inputs(run1.info.run_id, inputs_run1)
+    store.log_inputs(run2.info.run_id, inputs_run2)
+    store.log_inputs(run3.info.run_id, inputs_run3)
+
+    # Verify actual and expected results are same size and that all elements are equal.
+    def assert_has_same_elements(actual_list, expected_list):
+        assert len(actual_list) == len(expected_list)
+        for actual in actual_list:
+            # Verify the expected results list contains same element.
+            isEqual = False
+            for expected in expected_list:
+                isEqual = actual == expected
+                if isEqual:
+                    break
+            assert isEqual
+
+    # Verify no results from exp_id2 are returned.
+    results = store._search_datasets([exp_id1])
+    expected_results = [
+        DatasetSummary(exp_id1, dataset1.name, dataset1.digest, "train"),
+        DatasetSummary(exp_id1, dataset2.name, dataset2.digest, "eval"),
+        DatasetSummary(exp_id1, dataset2.name, dataset2.digest, "test"),
+    ]
+    assert_has_same_elements(results, expected_results)
```

Comment:
can we add a test case where one of the datasets does not have a context tag set?

Reply:
Yes, I added it into the existing tests.